firebase-perf/src/main/java/com/google/firebase/perf/FirebasePerfRegistrar.java

```diff
@@ -77,4 +89,10 @@ private static FirebasePerformance providesFirebasePerformance(ComponentContaine
 
     return component.getFirebasePerformance();
   }
+
+  @SuppressWarnings("FirebaseUseExplicitDependencies")
+  private static FirebasePerfEarly providesFirebasePerformanceEarly(ComponentContainer container) {
+    return new FirebasePerfEarly(
+        container.get(FirebaseApp.class), container.get(FirebaseApp.class).get(StartupTime.class));
```

Comment:
It's not recommended to get things from FirebaseApp, the error you are seeing wrt "undeclared dep" is due to the fact that you're declaring an `optionalProvider` dep but are then calling container.get() instead of `container.getProvider()`, so you should either switch to use `optional() + c.get()` or `optionalProvider() + c.getProvider()`